flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/catalog/FlinkCatalogManager.java

```diff
@@ -0,0 +1,152 @@
+	private String currentCatalogName;
+	private String currentDbName;
+
+	public FlinkCatalogManager() {
+		LOG.info("Initializing FlinkCatalogManager");
+		catalogs = new HashMap<>();
+
+		GenericInMemoryCatalog inMemoryCatalog = new GenericInMemoryCatalog(BUILTIN_CATALOG_NAME);
+		catalogs.put(BUILTIN_CATALOG_NAME, inMemoryCatalog);
+		currentCatalogName = BUILTIN_CATALOG_NAME;
+		currentDbName = inMemoryCatalog.getCurrentDatabase();
+
+		CatalogCalciteSchema.registerCatalog(rootSchema, BUILTIN_CATALOG_NAME, inMemoryCatalog);
+	}
+
+	@Override
+	public void registerCatalog(String catalogName, ReadableCatalog catalog) throws CatalogAlreadyExistException {
+		checkArgument(!StringUtils.isNullOrWhitespaceOnly(catalogName), "catalogName cannot be null or empty");
+		checkNotNull(catalog, "catalog cannot be null");
+
+		if (catalogs.containsKey(catalogName)) {
+			throw new CatalogAlreadyExistException(catalogName);
+		}
+
+		catalogs.put(catalogName, catalog);
+		catalog.open();
+		CatalogCalciteSchema.registerCatalog(rootSchema, catalogName, catalog);
+	}
+
+	@Override
+	public ReadableCatalog getCatalog(String catalogName) throws CatalogNotExistException {
+		if (!catalogs.keySet().contains(catalogName)) {
+			throw new CatalogNotExistException(catalogName);
+		}
+
+		return catalogs.get(catalogName);
+	}
+
+	@Override
+	public Set<String> getCatalogNames() {
+		return catalogs.keySet();
+	}
+
+	@Override
+	public ReadableCatalog getCurrentCatalog() {
+		return catalogs.get(currentCatalogName);
+	}
+
+	@Override
+	public String getCurrentDatabaseName() {
+		return currentDbName;
+	}
+
+	@Override
+	public void setCurrentCatalog(String catalogName) throws CatalogNotExistException {
+		checkArgument(!StringUtils.isNullOrWhitespaceOnly(catalogName), "catalogName cannot be null or empty");
+
+		if (!catalogs.keySet().contains(catalogName)) {
+			throw new CatalogNotExistException(catalogName);
+		}
+
+		if (!currentCatalogName.equals(catalogName)) {
+			currentCatalogName = catalogName;
+			currentDbName = catalogs.get(catalogName).getCurrentDatabase();
+			LOG.info("Set default catalog as '{}' and default database as '{}'", currentCatalogName, currentDbName);
+		}
+	}
+
+	@Override
+	public void setCurrentCatalogAndDatabase(String catalogName, String databaseName) throws CatalogNotExistException, DatabaseNotExistException {
+		checkArgument(!StringUtils.isNullOrWhitespaceOnly(catalogName), "catalogName cannot be null or empty");
+		checkArgument(!StringUtils.isNullOrWhitespaceOnly(databaseName), "databaseNName cannot be null or empty");
+
+		if (!catalogs.containsKey(catalogName)) {
+			throw new CatalogNotExistException(catalogName);
+		}
+
+		if (!catalogs.get(catalogName).listDatabases().contains(databaseName)) {
+			throw new DatabaseNotExistException(catalogName, databaseName);
+		}
+
+		if (!currentCatalogName.equals(catalogName) || !currentDbName.equals(databaseName)) {
+			currentCatalogName = catalogName;
```

Comment:
Shouldn't we set the database in the catalog?

Reply:
good catch. We kind of unified the "current database" concept in CatalogManager and the "default database" concept in a catalog in Flink, which are separated in Blink. I'll update accordingly